ADDRESSING.md

```diff
@@ -20,18 +20,20 @@
 
 ## TL;DR
 
-If no native protocol handler is available, redirect to a gateway. The implementation should detect if a local IPFS node is available. In web browser contexts where a local IPFS node is present, use [subdomain gateway](https://docs.ipfs.io/how-to/address-ipfs-on-web/#subdomain-gateway) at `localhost`. If not, use public one such as `dweb.link`:
+If no native protocol handler is available, redirect to IPFS path at a gateway. The implementation should detect if a local IPFS node is available. In web browser contexts where a local IPFS node is present, use [subdomain gateway](https://docs.ipfs.io/how-to/address-ipfs-on-web/#subdomain-gateway) at `localhost`. If not, use public one such as `dweb.link`. 
+
+In both cases point at IPFS path first, to ensure gateway takes care of CID normalization into a DNS-safe form:
 
 ```bash
-ipfs://{cid}                → https://{cid}.ipfs.dweb.link
-ipns://{libp2p-key}         → https://{libp2p-key}.ipns.dweb.link
+ipfs://{cid}          → https://dweb.link/ipfs/{cid}        → HTTP301 → https://{dns-safe-cid}.ipfs.dweb.link
+ipns://{libp2p-key}   → https://dweb.link/ipns/{libp2p-key} → HTTP301 → https://{dns-safe-cid}.ipns.dweb.link
```

Comment:
I like the clarity of the "HTTP301" but is not clear who is taking which action. I'd prefer that this somehow identify where the boundaries of the implementor's actions are, and where the gateway is.

Also, not required for now, but in the first arrows perhaps we should have a clarifying action like "expand to", like you did "HTTP301" to be clear about what's happening there.

Reply:
I've added abstract header in [`21a4fad` (#164)](https://github.com/ipfs/in-web-browsers/pull/164/commits/21a4fadbbc17cf407ab812d8eb1592a37f439a41) that clarifies the boundary a bit, but did not add "expand to", signal/noise ratio did not felt be beneficial:

```
Native URI            – HTTP Gateway                       (– Internal normalization done by HTTP Gateway)   
ipfs://{cid}          → https://dweb.link/ipfs/{cid}        → HTTP301 → https://{dns-safe-cid}.ipfs.dweb.link
```